### DISCUSSION TITLE
A quick and dirty OP3 visualiser in PHP

### DISCUSSION THREAD

**Discussion by jamescridland** [skymethod/op3#14] [Show and tell]
[This dirty Gist](https://gist.github.com/jamescridland/e982520faa5e5bc3cfcf926e768c3126) is a small PHP script that goes and fetches a day's worth of hits and displays "interesting" data from them.

Below is the output for all of Podnews's podcast episodes on 25 November 2022, as one example.

This is really awaiting the `/downloads/` API; but it's been an interesting thing to play with and see what happens. I've learnt quite a bit about how PHP handles arrays while writing it, so that's nice.

Interestingly, for comparison - [this is my logfile code](https://podnews.net/about/podcast-stats/526a574a-199b-439b-b0b0-2e033d18676a). My own code discards HEAD requests (and HEAD-like requests, like a one-byte range request) before it even reaches this. So, of the 2,771 OP3 "hits", my query has discarded 211 HEAD-like requests, and a further 89 bot requests. A known issue here is that if you grab my audio in three chunks, you'll appear here as three downloads.

<img width="1214" alt="Screen Shot 2022-11-29 at 12 31 38 pm" src="https://user-images.githubusercontent.com/231941/204424317-6937ff81-b084-46d9-a723-20acc88360ec.png">

**Comment by johnspurlock-skymethod** [comment]
Interesting - esp Indonesia
